### PR TITLE
BG/184252-Remove_excessive_retries

### PR DESCRIPTION
**Pull request by BarryGibney**
Initial commit - OSPlacesApiService changed in the web.config to disable retries

### CODE REVIEW

**Comment by BarryGibney** [comment]
The ticket suggested I use a disable feature for the retries, it didn't fully explain how this should be implemented, unfortunately I didn't get any questions answered - so this is how I have done it:

You can still use the retries, say 1,2,2,4
If you want one timeout for e.g 3 seconds without retrying you can do that ...so 3
If you want to disable the retries, this will then go to whatever default value is set

File: Web/Edubase.Services/IntegrationEndPoints/OSPlaces/OSPlacesApiService.cs [hunk 1]
```diff
@@ -25,8 +25,29 @@ public class OSPlacesApiService : IOSPlacesApiService
 
         private static Policy CreateRetryPolicy()
         {
-            var retryIntervalSettings = ConfigurationManager.AppSettings["OSPlacesApiServices_RetryIntervals"].Split(',');
-            var retryIntervals = retryIntervalSettings.Select(int.Parse).ToArray();
+            var defaultRetriesValue = new[] { 2 };
+            var retryEnabled = ConfigurationManager.AppSettings["OSPlacesApiServices_TimeoutIntervals_Enabled"] == "1";
+            var retryIntervalSettings = ConfigurationManager.AppSettings["OSPlacesApiServices_TimeoutIntervals"];
```
Comment (RogerHowellDfE) [inline Web/Edubase.Services/IntegrationEndPoints/OSPlaces/OSPlacesApiService.cs:30]: This behaviour is specifically about the intervals between retry attempts? 

If so, I'd suggest not including `TimeoutIntervals` in the name as this implies it is specifically about the timeout (how long the request is allowed to run for, as opposed to the gaps between the requests).

Similarly, whether the retries are enabled or not?

**Comment by RogerHowellDfE** [comment]
I'll follow up in chat with some other questions / comments / suggestions.

File: Web/Edubase.Services/IntegrationEndPoints/OSPlaces/OSPlacesApiService.cs (hunk 1 above)
Comment (anthony9853) [reply Web/Edubase.Services/IntegrationEndPoints/OSPlaces/OSPlacesApiService.cs:30]: Probably should be RetryIntervals.  Does this allow an empty set of intervals as no retries at all?

File: Web/Edubase.Services/IntegrationEndPoints/AzureMaps/AzureMapsService.cs [hunk 2]
```diff
@@ -26,12 +26,34 @@ public class AzureMapsService : IAzureMapsService
...
+            var retryIntervals = csvSeconds.Split(',')
+                .Select(x => x.Trim())
+                .Where(x => int.TryParse(x, out var seconds) && seconds >= 0)
+                .Select(x => TimeSpan.FromSeconds(int.Parse(x)))
+                .ToArray();
+
+            if (retryIntervals.Length == 0)
+            {
+                return new[] { TimeSpan.FromSeconds(2) };
+            }
```
Comment (RogerHowellDfE) [inline Web/Edubase.Services/IntegrationEndPoints/AzureMaps/AzureMapsService.cs:54]: For what reason are we defaulting to two attempts, with 2-seconds between the attempts?

Apologies if I've missed something with this!

Comment (BarryGibney) [reply Web/Edubase.Services/IntegrationEndPoints/AzureMaps/AzureMapsService.cs]: we shouldn't be - this should be applied to default timeouts not retries

**Review by RogerHowellDfE — APPROVED** [review]

**Pull request merged**